lib/ansible/modules/cloud/vmware/vsphere_file.py

```diff
@@ -0,0 +1,356 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2017, Dag Wieers (@dagwieers) <dag@wieers.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: vsphere_file
+short_description: Manage files on a vCenter datastore
+description:
+- Manage files on a vCenter datastore.
+version_added: '2.8'
+author:
+- Dag Wieers (@dagwieers)
+options:
+  host:
+    description:
+    - The vCenter server on which the datastore is available.
+    type: str
+    required: true
+    aliases: [ hostname ]
+  username:
+    description:
+    - The user name to authenticate on the vCenter server.
+    type: str
+    required: true
+  password:
+    description:
+    - The password to authenticate on the vCenter server.
+    type: str
+    required: true
+  datacenter:
+    description:
+    - The datacenter on the vCenter server that holds the datastore.
+    type: str
+    required: true
+  datastore:
+    description:
+    - The datastore on the vCenter server to push files to.
+    type: str
+    required: true
+  path:
+    description:
+    - The file or directory on the datastore on the vCenter server.
+    type: str
+    required: true
+    aliases: [ dest ]
+  validate_certs:
```

Comment:
Why are not we using `vmware` document fragments ?

Reply:
I'm guessing because this doesn't use the existing vmware utils

Reply:
It doesn't work on ESX, so the standard vmware doc fragments did not apply.